### PR TITLE
Add readLineIterator

### PR DESCRIPTION
**Pull request by marler8997**
Add the `readLineIterator` function to create a line iterator for a stream.

### CODE REVIEW

File: std/io.zig [hunk 1]
```diff
@@ -1067,6 +1067,51 @@ test "io.readLineFrom" {
...
 
+pub fn readLineIterator(stream: var, buf: *std.Buffer) type {
+    const StreamType = @typeOf(stream);
+    const Iterator = struct {
+        stream: StreamType,
+        buf: *std.Buffer,
+        pub fn next(self: *@This()) !?[]u8 {
+            return readLineFrom(self.stream, self.buf) catch |err| {
+                return switch (err) {
+                    error.EndOfStream => null,
```
Comment (daurnimator) [inline std/io.zig]: shouldn't this just return the final (unterminated) line?

Comment (marler8997) [reply std/io.zig]: In this case the stream has ended so there's no line to return.

Comment (daurnimator) [reply std/io.zig]: There is though: at least from my reading of the code, there's data left in `buf`.

Comment (marler8997) [reply std/io.zig]: There shouldn't be. Plus the test would fail if that was the case.

Comment (marler8997) [reply std/io.zig]: Unless, could readLineFrom return EndOfStream with data?

Comment (marler8997) [reply std/io.zig]: Looking through readLineFrom, it looks like you're right. Will have to push a fix.

**Comment by andrewrk** [comment]
I believe all the functionality here is provided by `readUntilDelimiterBuffer`. I don't see any additional benefits to this API.

Also the `std.io.readLine` API should probably be renamed; it is intended for the use case of e.g. a REPL. It's named after [GNU Readline](https://tiswww.cwru.edu/php/chet/readline/rltop.html). It's related to terminals, not streams.

**Pull request closed without merging**